src/Umbraco.Core/Services/MediaServiceExtensions.cs

```diff
@@ -17,7 +17,7 @@ namespace Umbraco.Extensions;
 /// </remarks>
 public static class MediaServiceExtensions
 {
-    public static IEnumerable<IMedia> GetByIds(this IMediaService mediaService, IEnumerable<Udi> ids)
+    public static IEnumerable<IMedia>? GetByIds(this IMediaService mediaService, IEnumerable<Udi> ids)
```

Comment:
`mediaService.GetByIds()` will never return null, it will return an empty `IEnumerable<IMedia>`, so marking this as nullable is redundant and just leads to extra unnecessary null checks. 

In fact, adding the `!`s elsewhere, points to the fact that this is an anti-pattern - they're added because we _know_ that the method can't return null... if we know it can't return null, then it why is it nullable?

```suggestion
    public static IEnumerable<IMedia> GetByIds(this IMediaService mediaService, IEnumerable<Udi> ids)
```